src/ddbToEs/index.ts

```diff
@@ -47,6 +46,20 @@ export async function handleDdbToEsEvent(event: any) {
 
         await ddbToEsHelper.logAndExecutePromises(promiseParamAndIds);
     } catch (e) {
-        console.log('Failed to update ES records', e);
+        console.error(
+            'Synchonization failed! The resources that could be effected are: ',
+            event.Records.map(
+                (record: {
+                    eventName: string;
+                    dynamodb: { OldImage: AWS.DynamoDB.AttributeMap; NewImage: AWS.DynamoDB.AttributeMap };
+                }) => {
+                    const image = record.eventName === REMOVE ? record.dynamodb.OldImage : record.dynamodb.NewImage;
+                    return `{id: ${image.id.S}, vid: ${image.vid.N}}`;
+                },
+            ),
+        );
+
+        console.error('Failed to update ES records', e);
+        throw e;
```

Comment:
I think that guaranteeing that only the failed messages go to the DLQ is a very desirable property of the system. Otherwise ops become harder for customers for no good reason (why are there so many DLQ messages? How come only 6% of them actually failed? How can I know which of them actually failed?)

Another desirable property is handling out of order messages. Our current implementation does not do that(not the same as idempotency). It could be achieved by updating ES only if the vid of the incoming message is higher than the vid of the document in ES. This would make it safe to redrive DLQ messages. I think we can tackle this later as a separate issue.

IMO sending only the failed messages to the DLQ should be done now (can still be a different PR). I agree that `BisectBatch` has scarce documentation, but is worth testing it out. Maybe `MaxRetry=4` and `BisectBatch=true` with our `BatchSize=15` will effectively isolate the error to a single record.

The cheap alternative is `MaxRetry=1`.
> I worry batch size of 1 may slow down our our sync too much

My intuition tells me the same, but we need data in order to discard that approach